### ISSUE TITLE
Incompatible types with esbuild 0.21.5

### ISSUE THREAD

**Issue by rowleyj** [badeball/cypress-cucumber-preprocessor#1198]
### Current behavior

```

 Property 'with' is missing in type 'import("<PATH>/node_modules/@badeball/cypress-cucumber-preprocessor/node_modules/esbuild/lib/main").OnResolveArgs' but required in type 'import("<PATH>/node_modules/esbuild/lib/main").OnResolveArgs'.
npm ERR! 
npm ERR! 27             plugins: [createEsbuildPlugin(config)],
```

### Desired behavior

Since `^0.21.2` is listed, we expect this package to work with all patch version of esbuild or restrict it to < `0.21.5`. Note that this is a brand [new version of esbuild](https://www.npmjs.com/package/esbuild?activeTab=versions).

### Test code to reproduce

This is not a functional issue, just a type issue that causes build errors. Setting esbuild to `0.21.3` resolved the issue, but logging as I anticipate others may run into this.

### Versions

* **Cypress version**:  "^13.6.6"
* **Preprocessor version**: "^19.2.0"
* **Node version**: 18.18.2

### Checklist

- [X] I've read the [FAQ](https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/docs/faq.md).
- [X] I've read [instructions for logging issues](https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I'm not using `cypress-cucumber-preprocessor@4.3.1` (package name has changed and it is no longer the most recent version, see [#689](https://github.com/badeball/cypress-cucumber-preprocessor/issues/689)).

**Comment by badeball** [comment]
The `esbuild-ts` [example](https://github.com/badeball/cypress-cucumber-preprocessor/tree/master/examples/esbuild-ts), which uses all latests versions, works seemingly fine. You need to provide a reproducible example to your issue.

**Comment by hishchuk** [comment]
Hi!
I have the same issue
![image](https://github.com/badeball/cypress-cucumber-preprocessor/assets/149797091/b436ef93-6d7c-4347-a77d-92e7cfefdaba).
It looks there is kind of inconsistency. As a temporary solution I'm adding highlighted line below. Please let me know if more details are needed.
![image](https://github.com/badeball/cypress-cucumber-preprocessor/assets/149797091/fbe252b2-278b-4fd6-ab3a-9111dcab6168)

**Comment by badeball** [comment]
> Please let me know if more details are needed.

Alright - your post doesn't provide any useful information to this issue. If you want to see this issue resolved and contribute to that, you can provide a [reproducible example](https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/CONTRIBUTING.md#reduced-test-case), like OP has been asked of twice now: one in the issue template, twice in the comments.

**Comment by hishchuk** [comment]
Hi,

In your [example](https://github.com/badeball/cypress-cucumber-preprocessor/tree/master/examples/esbuild-ts) could you please just use attached tsconfig file.
[tsconfig.json](https://github.com/user-attachments/files/15834769/tsconfig.json)

Unfortunately I cannot change moduleResolution property, so [workaround](https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/docs/faq.md#i-get-cannot-find-module-badeballcypress-cucumber-preprocessoresbuild) was used.
I guess this is the main reason of the problem.

Thank you

**Comment by badeball** [comment]
I've updated dependencies, including `esbuid`, which I believe fixes this, released as v20.0.7. Try it out.

**Comment by hishchuk** [comment]
Everything is fine now. Thanks a lot!
